lib/puppet/resource_api.rb

```diff
@@ -341,7 +341,7 @@ def call_provider(value); end
       end
 
       define_singleton_method(:context) do
-        @context ||= PuppetContext.new(definition[:name])
+        @context ||= PuppetContext.new(definition[:name], definition)
```

Comment:
This can be tightened up to only pass the definition, and have the name as a `Type` attr_reader.